web_ui/frontend/app/registry/components/CustomRegistrationField/EnumerationField.tsx

```diff
@@ -0,0 +1,37 @@
+import {Autocomplete, TextField} from "@mui/material";
+import React, {useMemo} from "react";
+
+import type {CustomRegistrationFieldProps} from "./index.d";
+
+
+const EnumerationField = ({onChange, displayed_name, name, required, description, value, options}: CustomRegistrationFieldProps<string>) => {
+
+    const textValue = useMemo(() => options?.find(option => option.id === value), [value, options])
+
+    return <Autocomplete
+        fullWidth
+        size="small"
+        renderInput={(params) => {
+            return <TextField
+                {...params}
+                required={required}
+                helperText={description}
+                label={displayed_name}
+                variant={"outlined"}
+            />
+        }}
+        onChange={(e, value) => {
+            const enumerateId = options?.find(option => option.name === value?.name)?.id
```

Comment:
The names of the options are not unique AFAIK (at least we don't check them), is there a way to get use ID as the value instead?

Reply:
@haoming29 I think we should make the names unique then. If the names aren't unique how is a user supposed to distinguish between 'x' and 'x'?

Reply:
I'd offer adding the id to the ui but the ID is likely just Primary Key hash.

Reply:
Since this is a bit of a edge case, that won't break anything if it hits ( first one would just get picked ) I say we move forward on this PR.

Reply:
@haoming29 Just tested and the web ui prevents duplicate names. So this is even more of an edge case.

Reply:
Yeah I think it doesn't make sense to have duplicated item names and it sounds even better if the UI can handle this case

Reply:
@haoming29 Sounds good, I will make a ticket to have the config verifier check for unique names and merge this.